src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.ts

```diff
@@ -0,0 +1,147 @@
+    public rbdConfigurationService: RbdConfigurationService
+  ) {}
+
+  ngOnInit() {
+    const configFormGroup = this.createConfigurationFormGroup();
+    this.form.addControl('configuration', configFormGroup);
+
+    // Listen to changes and emit the values to the parent component
+    configFormGroup.valueChanges.subscribe(() => {
+      this.changes.emit(this.getDirtyValues.bind(this));
+    });
+
+    if (this.initializeData) {
+      this.initializeData.subscribe((data) => {
+        this.initialData = data.initialData;
+        const dataType = data.sourceType;
+
+        this.rbdConfigurationService.getWritableOptionFields().forEach((option) => {
+          const optionData = data.initialData.filter((entry) => entry.name === option.name).pop();
+          if (optionData && optionData['source'] === dataType) {
+            this.form.get(`configuration.${option.name}`).setValue(optionData['value']);
+          }
+        });
+        this.ngDataReady.emit();
+      });
+    }
+
+    this.rbdConfigurationService
+      .getWritableSections()
+      .forEach((section) => (this.sectionVisibility[section.class] = false));
+  }
+
+  getDirtyValues(includeLocalValues = false, localFieldType?: RbdConfigurationSourceField) {
+    const result = {};
+
+    this.rbdConfigurationService.getWritableOptionFields().forEach((config) => {
+      const control = this.form.get('configuration').get(config.name);
+      const dirty = control.dirty;
+
+      if (this.initialData && this.initialData[config.name] === control.value) {
+        return; // Skip controls with initial data loaded
+      }
+
+      if (dirty || (includeLocalValues && control['source'] === localFieldType)) {
+        if (control.value === null) {
+          result[config.name] = control.value;
+        } else if (config.type === RbdConfigurationType.bps) {
+          result[config.name] = this.formatterService.toBytes(control.value);
+        } else if (config.type === RbdConfigurationType.milliseconds) {
+          result[config.name] = this.formatterService.toMilliseconds(control.value);
+        } else if (config.type === RbdConfigurationType.iops) {
+          result[config.name] = this.formatterService.toIops(control.value);
+        } else {
+          result[config.name] = control.value;
+        }
+      }
+    });
+
+    return result;
+  }
+
+  /**
+   * Dynamically create form controls.
+   */
+  private createConfigurationFormGroup() {
+    const configFormGroup = new CdFormGroup({});
+
+    this.rbdConfigurationService.getWritableOptionFields().forEach((c) => {
+      let control: FormControl;
+      if (
+        c.type === RbdConfigurationType.milliseconds ||
+        c.type === RbdConfigurationType.iops ||
+        c.type === RbdConfigurationType.bps
+      ) {
+        control = new FormControl(0, Validators.min(0));
+      } else {
+        throw new Error(
+          `Type ${c.type} is unknown, you may need to add it to RbdConfiguration class`
```

Comment:
```suggestion
          this.i18n('Type {{type}} is unknown, you may need to add it to RbdConfiguration class', { type: c.type });
```

Reply:
Do we really want or need to translate a developer-only error message?